parastell/invessel_build.py

```diff
@@ -49,134 +53,99 @@ def orient_spline_surfaces(volume_id):
-            If none is supplied, 'plasma' will be used (defaults to None).
-        sol_mat_tag (str): alternate DAGMC material tag to use for scrape-off
-            layer. If none is supplied, 'sol' will be used (defaults to None).
-        logger (object): logger object (defaults to None). If no logger is
-            supplied, a default logger will be instantiated.
     """
 
     def __init__(
-            self,
-            vmec,
-            toroidal_angles,
-            poloidal_angles,
-            radial_build,
-            wall_s,
-            repeat=0,
-            num_ribs=61,
-            num_rib_pts=61,
-            scale=m2cm,
-            plasma_mat_tag=None,
-            sol_mat_tag=None,
-            logger=None
+        self,
+        vmec_obj,
+        radial_build,
+        logger=None,
+        **kwargs
     ):
-        self.vmec = vmec
-        self.toroidal_angles = toroidal_angles
-        self.poloidal_angles = poloidal_angles
-        self.radial_build = radial_build
-        self.wall_s = wall_s
-        self.repeat = repeat
-        self.num_ribs = num_ribs
-        self.num_rib_pts = num_rib_pts
-        self.scale = scale
-        self.plasma_mat_tag = plasma_mat_tag
-        self.sol_mat_tag = sol_mat_tag
 
         self.logger = logger
-        if self.logger == None or not self.logger.hasHandlers():
-            self.logger = log.init()
+        self.vmec_obj = vmec_obj
+        self.radial_build = radial_build
+        
+        self.repeat = 0
+        self.num_ribs = 61
+        self.num_rib_pts = 67
+        self.scale = m2cm
+
+        allowed_kwargs = ['repeat', 'num_ribs', 'num_rib_pts', 'scale']
+        set_kwarg_attrs(
+            self,
+            kwargs,
+            allowed_kwargs,
+            self._logger
+        )
 
         self.Surfaces = {}
         self.Components = {}
 
-        if (self.repeat + 1) * self.toroidal_angles[-1] > 360.0:
+    @property
+    def vmec_obj(self):
+        return self._vmec_obj
+    
+    @vmec_obj.setter
+    def vmec_obj(self, vmec_object):
+        self._vmec_obj = vmec_object
+
+    @property
+    def logger(self):
+        return self._logger
+    
+    @logger.setter
+    def logger(self, logger_object):
+        self._logger = logger_object
+        if self._logger == None or not self._logger.hasHandlers():
+            self._logger = log.init()
+        
+    @property
+    def repeat(self):
+        return self._repeat
+    
+    @repeat.setter
+    def repeat(self, num):
+        self._repeat = num
+        if (self._repeat + 1) * self.radial_build.toroidal_angles[-1] > 360.0:
             e = AssertionError(
                 'Total toroidal extent requested with repeated geometry '
-                'exceeds 360 degrees. Please examine phi_list and the repeat '
-                'parameter.'
+                'exceeds 360 degrees. Please examine the \'repeat\' parameter '
+                'and the \'toroidal_angles\' parameter of \'radial_build\'.'
```

Comment:
You can avoid having to escape these single quotes by using the other quote mark from that defining the string
```suggestion
                'exceeds 360 degrees. Please examine the "repeat" parameter '
                'and the "toroidal_angles" parameter of "radial_build".'
```